src/app/modules/shared_models/message_model.nim

```diff
@@ -352,6 +352,26 @@ QtObject:
 
     return self.items[ind]
 
+  proc itemSent*(self: Model, messageId: string) =
+    let ind = self.findIndexForMessageId(messageId)
+    if(ind == -1):
+      return
+
+    self.items[ind].outgoingStatus = "send"
+
+    let index = self.createIndex(ind, 0, nil)
+    self.dataChanged(index, index, @[ModelRole.OutgoingStatus.int])
+
+  proc itemDelivered*(self: Model, messageId: string) =
```

Comment:
@faust4exe since both methods refer to the same prop which need to be updated you can optimize the code a bit using e.g. `setOutgoingStatus` private proc:
```
proc setOutgoingStatus(self: Model, messageId: string, status: string) =
  let ind = self.findIndexForMessageId(messageId)
  if(ind == -1):
    return
  self.items[ind].outgoingStatus = status
  let index = self.createIndex(ind, 0, nil)
  self.dataChanged(index, index, @[ModelRole.OutgoingStatus.int])

proc markMessageSent*(self: Model, messageId: string) =
  self.setOutgoingStatus(messageId, MSG_OUTGOING_STATUS_SEND)

proc markMessageDelivered*(self: Model, messageId: string) =
  self.setOutgoingStatus(messageId, MSG_OUTGOING_STATUS_DELIVERED)
```

Reply:
Ok - will update soon.